### PR TITLE
Fix spawn effect

### PR DESCRIPTION
**Pull request by BertovDev**
Closes #[issue]

## Motivation

Currently the spawn effect was desactivated. With this we activate and desactive it for all players in the init of the game.

## Summary of changes

Changed CharacterBase activateSpawnFeedback method to ToggleSpawnFeedback
Re-created the logic activation in CustomLevelManager `CameraCinematic` method

## How has this been tested?

Start a game you should see the spawnfeedback once at that time and not ever again.

## Test additions / changes

List tests added/updated.

## Checklist
- [x] I have tested the changes locally.
- [x] I self-reviewed the changes on GitHub, line by line.
- [ ] Tests have been added/updated.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.
- [x] I have tested the changes in another devices.
  - [ ] Tested in iOS.
  - [x] Tested in Android.

### CODE REVIEW

File: client/Assets/Scripts/CharacterBase.cs [hunk 1]
```diff
@@ -27,21 +27,19 @@ public class CharacterBase : MonoBehaviour
...
+        //     MMSoundManagerSoundPlayEvent.Trigger(
+        //         spawnSfx,
+        //         MMSoundManager.MMSoundManagerTracks.Sfx,
+        //         Utils.GetPlayer(SocketConnectionManager.Instance.playerId).transform.position
+        //     );
+        // } else {
+        //     sound3DManager.SetSfxSound(spawnSfx);
+        //     sound3DManager.PlaySfxSound();
+        // }
     }
```
Comment (agustinesco) [inline client/Assets/Scripts/CharacterBase.cs:44]: Why are we keeping this code if we're not using it anymore?

File: client/Assets/Scripts/CustomLevelManager.cs [hunk 2]
```diff
@@ -196,6 +201,19 @@ IEnumerator CameraCinematic()
             yield return new WaitForSeconds(2.1f);
             CancelInvoke("Substract");
             InvokeRepeating("MoveYCamera", 0.3f, 0.1f);
+            Utils
+                .GetAllCharacters()
+                .ForEach(character =>
+                {
+                    character.characterBase.ToggleSpawnFeedback(true, character.PlayerID);
```
Comment (agustinesco) [inline client/Assets/Scripts/CustomLevelManager.cs:208]: Why not changing the `spawnFeedback` here?
```suggestion
                    character.characterBase.spawnFeedback.SetActive(true);
```

The same goes for the function call below

**Review by agustinesco — CHANGES_REQUESTED** [review]
The spawn effect seems to be working as expected! nice.
I just have 1 question and some comments regarding the code.

With this change we are losing the spawn sounds, is this okey?

File: client/Assets/Scripts/CharacterBase.cs [hunk 3]
```diff
@@ -22,24 +22,18 @@ public class CharacterBase : MonoBehaviour
     [SerializeField]
     Sound3DManager sound3DManager;
 
-    public IEnumerator activateSpawnFeedback(bool isCurrentPlayer)
+    public void ToggleSpawnFeedback(bool value, string id)
```
Comment (agustinesco) [inline client/Assets/Scripts/CharacterBase.cs:25]: Can we rename the var value to something that indicate what we're receiving?? something like `activateSound` or `activateFeedback`

**Review by agustinesco — CHANGES_REQUESTED** [review]
The sounds seems to be working as expected but whenever i enter a match i get this error
image

File: client/Assets/Scripts/CharacterBase.cs (hunk 3 above)
Comment (BertovDev) [reply client/Assets/Scripts/CharacterBase.cs:25]: NIce catch, i will change it.

**Comment by BertovDev** [comment]
> The sounds seems to be working as expected but whenever i enter a match i get this error image

This message is totally expected. It happends beucause in this PR I changed the audio listener from the camera to the player, so until the characters are not created you will have this message send. Nothing to worry about and good that you notice it,

**Comment by agustinesco** [comment]
> > The sounds seems to be working as expected but whenever i enter a match i get this error image
> 
> This message is totally expected. It happends beucause in this PR I changed the audio listener from the camera to the player, so until the characters are not created you will have this message send. Nothing to worry about and good that you notice it,

After talking about this we think that this is only a message and we should ignore it, you can read more about it [here](https://forum.unity.com/threads/there-are-no-audio-listeners-in-the-scene-thats-actually-fine-right.384596/)

**Review by agustinesco — DISMISSED** [review]

File: client/Assets/Scripts/CharacterBase.cs [hunk 4]
```diff
@@ -22,24 +22,18 @@ public class CharacterBase : MonoBehaviour
...
+        spawnFeedback.SetActive(isActiveSound);
+        if (isActiveSound)
         {
             MMSoundManagerSoundPlayEvent.Trigger(
                 spawnSfx,
                 MMSoundManager.MMSoundManagerTracks.Sfx,
-                Utils.GetPlayer(SocketConnectionManager.Instance.playerId).transform.position
+                Utils.GetPlayer(SocketConnectionManager.Instance.playerId).transform.position,
+                false,
+                0.01f
```
Comment (ncontinanza) [inline client/Assets/Scripts/CharacterBase.cs:35]: We could have this value as a private constant, such as we did with the minimum volume in ToggleAudio.cs

**Review by ncontinanza — CHANGES_REQUESTED** [review]

**Review by ncontinanza — COMMENTED** [review]
Also, there's a cut in the game music during the "preparing battle" screen. The music stops, and after this screen it keeps on playing were it left

File: client/Assets/Scripts/CharacterBase.cs (hunk 4 above)
Comment (BertovDev) [reply client/Assets/Scripts/CharacterBase.cs]: Done!

**Review by ncontinanza — APPROVED** [review]
LGTM!

**Review by agustinesco — APPROVED** [review]
LGTM!

**Pull request merged**